src/App.tsx

```diff
@@ -21,55 +20,140 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
+type State = {
+  goods: string[];
+  isVisible: boolean;
   isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  isSorted: boolean,
+  sortBy: SortType,
+};
 
-  // Sort and reverse goods if needed
-  // ...
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    isSorted: false,
+    sortBy: SortType.NONE,
+  };
 
-  return visibleGoods;
-}
+  visibleGoodsList = () => {
+    this.setState((state) => ({
+      isVisible: !state.isVisible,
+    }));
+  };
 
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState(() => ({
+      sortBy: SortType.NONE,
+      isReversed: false,
+      isSorted: false,
+    }));
+  };
+
+  sortByHandler = (sortBy: SortType) => {
+    this.setState({
+      isSorted: true,
+      sortBy,
+    });
+  };
+
+  render() {
+    const {
+      isVisible,
+      goods,
+      isReversed,
+      sortBy,
+      isSorted,
+    } = this.state;
+
+    const newGoods = [...goods];
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+    if (isSorted) {
+      newGoods.sort((good1, good2): number => {
+        switch (sortBy) {
+          case SortType.LENGTH:
+            return good1.length - good2.length;
+
+          case SortType.ALPABET:
+            return good1.localeCompare(good2);
+
+          default: return 0;
+        }
+      });
+    }
+
+    if (isReversed) {
+      newGoods.reverse();
+    }
```

Comment:
this logic is for a separate function
just to explain you why: combining logic of several separate processes in one method (in this case – `render`) is not a common practice, it makes your unmaintainable

Reply:
strongly recommend you using this approach next times